util.go

```diff
@@ -27,3 +28,11 @@ func randomBytes(n int) []byte {
 	}
 	return rv
 }
+
```

Comment:
nit: I think this is better suited to being in `samlsp/` than the root package.

Reply:
I would have done that, but samlidp makes use of it too: https://github.com/crewjam/saml/pull/116/files#diff-2748fea7ed8e13ed1d66d922ba24dbc2

Do you still think it should be moved there?

Reply:
it is so short, maybe the thing to do is to make it private in both packages. What I'd prefer not to do is have it be part of the public interface of the root package. :)

Reply:
It's simple enough that I like this way: https://github.com/crewjam/saml/pull/116/commits/27a74e8090487fb060745fda343620935f5cf511